app/src/main/java/co/stonephone/stonecamera/plugins/QRScanner.kt

```diff
@@ -133,57 +133,81 @@ class QRScannerPlugin : IPlugin {
-        val deferred = CompletableDeferred<Unit>()
-
-        val inputImage =
-            InputImage.fromMediaImage(image, imageProxy.imageInfo.rotationDegrees)
-
-        scanner?.process(inputImage)
-            ?.addOnSuccessListener { barcodes ->
-                if (barcodes.size > 0) {
-                    val prevBarcode = value
-                    val barcode = barcodes[0]
-                    value = barcode.rawValue
-
-                    if (value !== prevBarcode) {
-                        // Trigger haptic feedback
-                        val vibrator = MyApplication.getAppContext()
-                            .getSystemService(Context.VIBRATOR_SERVICE) as Vibrator
-                        vibrator.vibrate(
-                            VibrationEffect.createOneShot(
-                                100,
-                                VibrationEffect.DEFAULT_AMPLITUDE
+        var deferred = CompletableDeferred<Unit>()
+
+        if (qrScannerEnabled == "ON") {
+            val inputImage =
+                InputImage.fromMediaImage(image, imageProxy.imageInfo.rotationDegrees)
+
+            scanner?.process(inputImage)
+                ?.addOnSuccessListener { barcodes ->
+                    if (barcodes.size > 0) {
+                        val prevBarcode = value
+                        val barcode = barcodes[0]
+                        value = barcode.rawValue
+
+                        if (value !== prevBarcode) {
+                            // Trigger haptic feedback
+                            val vibrator = MyApplication.getAppContext()
+                                .getSystemService(Context.VIBRATOR_SERVICE) as Vibrator
+                            vibrator.vibrate(
+                                VibrationEffect.createOneShot(
+                                    100,
+                                    VibrationEffect.DEFAULT_AMPLITUDE
+                                )
                             )
-                        )
-                    }
+                        }
 
-                    // Cancel any pending clear
-                    clearValueRunnable?.let { handler.removeCallbacks(it) }
-                } else {
-                    if (value !== null && clearValueRunnable == null) {
-                        clearValueRunnable = Runnable {
-                            value = null
-                            barcodePos = null
-                            clearValueRunnable = null
+                        // Cancel any pending clear
+                        clearValueRunnable?.let { handler.removeCallbacks(it) }
+                    } else {
+                        if (value !== null && clearValueRunnable == null) {
+                            clearValueRunnable = Runnable {
+                                value = null
+                                clearValueRunnable = null
+                            }
+                            handler.postDelayed(clearValueRunnable!!, 5000)
```

Comment:
As part of making the code customisable for various manufacturers to use, how useful would it be to make these sorts of values easy to specify/modify (rather than someone searching through all the code and then editing values as they find them)?